bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+            localServiceRegistration.unregister();
+            serviceRegistration = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            logger.debug("Unsupported refresh command: {}", command);
+        } else {
+            switch (channelUID.getId()) {
+                case POWER:
+                    debounce = false;
+                    synchronized (sequenceLock) {
+                        try {
+                            if (command == OnOffType.ON) {
+                                communicator.sendKeyPress(KeyCommand.POWERON.getJson());
+                                powerOn = true;
+                            } else {
+                                communicator.sendKeyPress(KeyCommand.POWEROFF.getJson());
+                                powerOn = false;
+                            }
+                        } catch (VizioException e) {
+                            logger.debug("Unable to send power {} command to the Vizio TV, Exception: {}", command,
+                                    e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        }
+                    }
+                    break;
+                case VOLUME:
+                    debounce = false;
+                    synchronized (sequenceLock) {
+                        try {
+                            int volume = Integer.parseInt(command.toString());
+
+                            // volume changed again before polling has run, get current volume hash from the TV first
+                            if (currentVolumeHash.equals(0L)) {
+                                Audio audioSettings = communicator.getCurrentAudioSettings();
+
+                                Optional<ItemAudio> volumeItem = audioSettings.getItems().stream()
+                                        .filter(i -> VOLUME.equals(i.getCname())).findFirst();
+                                if (volumeItem.isPresent()) {
+                                    currentVolumeHash = volumeItem.get().getHashval();
+                                } else {
+                                    logger.debug("Unable to get current volume hash on the Vizio TV");
+                                }
+                            }
+                            communicator
+                                    .changeVolume(String.format(MODIFY_INT_SETTING_JSON, volume, currentVolumeHash));
+                            currentVolumeHash = 0L;
+                        } catch (VizioException e) {
+                            logger.debug("Unable to set volume on the Vizio TV, command volume: {}, Exception: {}",
+                                    command, e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Consider if the message above or even the exception message should be included in this status update to provide more immediate information to the user.